solution_collector/src/main.rs

```diff
@@ -12,85 +14,135 @@ use github_rs::client::{Github, Executor};
 use serde_json::Value;
 use regex::Regex;
 use std::collections::HashMap;
+use structopt::StructOpt;
 
-#[derive(Debug)]
-enum Mode {
+#[derive(StructOpt, Debug)]
+#[structopt(name = "solution_collector")]
+struct Opt {
+    #[structopt(subcommand)]
+    command: Command
+}
+
+#[derive(StructOpt, Debug)]
+enum Command {
+    #[structopt(name = "preview")]
     Preview,
-    Merge
+    #[structopt(name = "week-report")]
+    WeekReport {
+        week: u32
+    },
+    #[structopt(name = "merge")]
+    Merge {
+        week: u32,
```

Comment:
I don't see any logical difference between `WeekReport::week` and `Merge::week`. Aren't they supposed to be the same?

Reply:
@huytd: week report is for, well, week report 😂. It will only show that week’s submissions, compare them against PARTICIPANTS list, group submitted entries by number of solved problem.